### DISCUSSION TITLE
if you use ethers-to-web3？

### DISCUSSION THREAD

**Discussion by QianDuoDuo-web** [WalletConnect/walletconnect-monorepo#3468] [Developer Support] [answered]
web3Model 2.7.1 version, if you use ethers-to-web3, switch to web3.js program, I am urgent, I need help.

**Comment by glitch-txs** [comment, marked as answer]
Web3.js is currently not directly supported. You can try using Wagmi adapter and replace ethers with web3js
https://wagmi.sh/react/ethers-adapters

We're adding support for web3.js in Web3Modal v3

**Reply by QianDuoDuo-web** [reply]
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/51685986/63f48316-7239-4f66-96a0-280c7b63fad0)
Hello, I can't access this link, is there a code snippet for my reference? i need it

**Reply by glitch-txs** [reply]
Look for wagmi.sh website :)

**Reply by QianDuoDuo-web** [reply]
import { configureChains, createConfig } from "@wagmi/core";
import { arbitrum, avalanche, mainnet, polygon } from "@wagmi/core/chains";
import {
  EthereumClient,
  w3mConnectors,
  w3mProvider,
} from "@web3modal/ethereum";
import { Web3Modal } from "@web3modal/html";

// 1. Define constants
const projectId = import.meta.env.VITE_PROJECT_ID;
if (!projectId) {
  throw new Error("You need to provide VITE_PROJECT_ID env variable");
}

const chains = [mainnet, polygon, avalanche, arbitrum];

// 2. Configure wagmi client
const { publicClient } = configureChains(chains, [w3mProvider({ projectId })]);
const wagmiConfig = createConfig({
  autoConnect: true,
  connectors: w3mConnectors({ chains, version: 1, projectId }),
  publicClient,
});

// 3. Create ethereum and modal clients
const ethereumClient = new EthereumClient(wagmiConfig, chains);
export const web3Modal = new Web3Modal(
  {
    projectId,
    walletImages: {
      safe: "https://pbs.twimg.com/profile_images/1566773491764023297/IvmCdGnM_400x400.jpg",
    },
  },
  ethereumClient
);



I just read this document and found that I don't know where to start to change this code
